src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,146 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private static class Node<T> {
+        private T data;
+        private Node<T> prev;
+        private Node<T> next;
+
+        Node(T data, Node<T> prev, Node<T> next) {
+            this.data = data;
+            this.prev = prev;
+            this.next = next;
+        }
+    }
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+        head = null;
+        tail = null;
+        size = 0;
+    }
+
+    private void link(Node<T> prev, Node<T> newNode, Node<T> next) {
```

Comment:
![image](https://github.com/mate-academy/jv-linked-list/assets/20326858/3b9b0ba5-235f-44ba-8cf5-27393047dc04)
```suggestion
    private void linkBefore(Node<T> newNode, Node<T> next) {
```